beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -233,8 +233,11 @@ func (f *blocksFetcher) handleRequest(ctx context.Context, start, count uint64)
 	}
 
 	headEpoch := helpers.SlotToEpoch(f.headFetcher.HeadSlot())
-	_, finalizedEpoch, peers := f.p2p.Peers().BestFinalized(params.BeaconConfig().MaxPeersToSync, headEpoch)
-
+	_, finalizedEpoch, peers, err := f.p2p.Peers().BestFinalized(params.BeaconConfig().MaxPeersToSync, headEpoch)
+	if err != nil {
+		response.err = err
+		return response
+	}
```

Comment:
Code is checking the `len(pids)` just on the next line - not sure if moving that check to `BestFinalized` is motivated enough.